src/lib/lore.sh

```diff
@@ -411,82 +411,52 @@ function get_patches_from_mailing_list()
   done
 }
 
-function title_to_path()
-{
-  local title="$1"
-  local file_name
-
-  # Replace space in favor of _
-  file_name="${title// /_}"
-
-  # Replace .
-  file_name="${file_name//./_}"
-
-  # Replace special character
-  file_name="${file_name//[&*+%!:,]/}"
-
-  # Replace /
-  file_name="${file_name//[\/\\]/-}"
-
-  printf '%s' "$file_name"
-}
-
-function convert_title_to_patch_name()
-{
-  local title="$1"
-  local index="$2"
-  local file_name
-
-  file_name=$(title_to_path "$title")
-
-  if [[ -n "$index" ]]; then
-    printf '%04d-%s\n' "$index" "${file_name}.mbox"
-    return
-  fi
-
-  printf '%s\n' "${file_name}.mbox"
-}
-
-function url_update_patch_number()
-{
-  local url="$1"
-  local new_number="$2"
-
-  url="${url/-[0-9]*-/-$link_ref-}"
-
-  printf '%s' "$url"
-}
-
+# This function downloads a patch series thread in a .mbox format to a given directory
+# using a series URL. The series URL should be the concatenation of the lore url, the
+# target mailing list, and the message ID. Below is an example of such a URL:
+#   https://lore.kernel.org/some-list/2367462342.4535535-1-email@email.com/
+# The output filename is '<message ID>.mbox'. This function uses the `b4` tool underneath to
+# delegate the process of fetching and downloading the thread.
+#
+# Return:
+# Return 0 if the thread was successfully downloaded, 22 if the series URL or the output
+# directory passed as arguments is empty and the error code of `b4` in case it fails.
 function download_series()
 {
-  local total_patches="$1"
-  local first_message_id="$2"
-  local save_to="$3"
-  local title="$4"
-  local total=0
-  local link_ref=0
+  local series_url="$1"
+  local save_to="$2"
+  local flag=${3:-'SILENT'}
   local url
-  local patch_file_name
+  local series_filename
+  local cmd
+  local ret
 
-  mkdir -p "$save_to"
+  # Safety checking
+  if [[ -z "$series_url" || -z "$save_to" ]]; then
+    return 22 # EINVAL
+  fi
 
-  url=$(replace_http_by_https "$first_message_id")
+  # Create the output directory if it doesn't exists
+  cmd_manager "$flag" "mkdir --parents '${save_to}'"
+  ret="$?"
+  [[ "$ret" != 0 ]] && return "$ret"
 
-  until ! is_the_link_valid "$url"; do
-    ((total++))
-    ((link_ref++))
+  # Prepare filename and url for downloading patches
+  series_filename=$(printf '%s' "$series_url" | rev | cut --delimiter '/' -f2 | rev)
+  url=$(replace_http_by_https "$series_url")
 
-    url=$(url_update_patch_number "$url" "$link_ref")
-    patch_file_name=$(convert_title_to_patch_name "$title" "$link_ref")
-    download "${url}raw" "$patch_file_name" "${save_to}" &
-  done
-  wait
+  # Issue the command to download the series
+  cmd="b4 --quiet mbox '${url}' --outdir '${save_to}' --mbox-name '${series_filename}.mbox'"
```

Comment:
@rodrigosiqueira the `mbox` subcommand results in the downloading of the whole thread. If we use `am` instead, everything not essential for using `git am` is trimmed and the patches are ordered. The previous version of `download_series` practically used this second approach, but:
1. It didn't differ the cover letter from the patches (resulted in errors)
2. The downloaded file was broken down into `n` files (`n` being the number of patches + cover letter).

What do you think?